pre-cache/olm

```diff
@@ -62,37 +63,36 @@ olm_main(){
     for index in $(sort -u $config_volume_path/operators.indexes) ; do
 
         image_id=$(pull_index $index $pull_secret_path)
-        rv=$? && ret_val=ret_val+$rv
-        [[ $rv -eq 0 ]] || continue
+        if [[ $? -ne 0 ]]; then
+          log_debug "pull_index failed for index $index"
+          return 1
+        fi
         log_debug "$index image ID is $image_id"
         image_mount=$(mount_index $image_id)
-        rv=$? && ret_val=ret_val+$rv
-        [[ $rv -eq 0 ]] || continue
+        if [[ $? -ne 0 ]]; then
+          log_debug "mount_index failed for index $index"
+          return 1
+        fi
         log_debug "Image mount: $image_mount"
-        
         packages=$(extract_packages)
         if ! [[ -n $packages ]]; then
-          log_debug "Operators index is set, but no packages provided - inconsistent configuration"
+          log_debug "operators index is set, but no packages provided - inconsistent configuration"
           return 1
         fi
-        export_index $index $pull_secret_path $packages $image_mount
+        render_index $index $pull_secret_path $packages $image_mount
         if [[ $? -ne 0 ]]; then
-          log_debug "export_index failed: OLM index export failed for index $index, package(s) $packages"
+          log_debug "render_index failed: OLM index render failed for index $index, package(s) $packages"
           return 1
         fi
-
         operators_spec_file="$config_volume_path/operators.packagesAndChannels"
-        extract_pull_spec $index_download_path $operators_spec_file $pull_spec_file
+        extract_pull_spec $rendered_index_path $operators_spec_file $pull_spec_file
         if [[ $? -ne 0 ]]; then
           log_debug "extract_pull_spec failed"
           return 1
         fi
         unmount_index $image_id
     done
-    if [[ $ret_val -ne 0 ]]; then
-        log_debug "Some of the OLM index extractions failed."
-    fi
-    return $ret_val
+    return 0
```

Comment:
not sure if this is needed, if loop finishes then the implicit return will be the last command of the function, so the return code should be implicitly zero I believe.